### DISCUSSION TITLE
Convert collection

### DISCUSSION THREAD

**Discussion by gabrielgrazziani** [Kong/insomnia#3486] [Help] [answered]
Is it possible to convert collection to document?

**Comment by reynolek** [comment]
Hi @gabrielgrazziani -- I am not sure that this would be very beneficial. In a Document, the debug tab is powered by the OpenAPI spec. If you don't have a spec ready to go, you would lose all those requests in your collection. If you have an OpenAPI spec, it would be more beneficial to just create a new document and use that spec.

Does that make sense? Or am I not understanding your ask?

**Comment by develohpanda** [comment, marked as answer]
Hi! May I ask for what purpose you are looking to switch from collection to document? Is it for unit tests, for the openapi spec, or to generate an openapi spec from the collection of requests, or something else?

Anyway, short answer is: no you can't convert right now. You also can't, unfortunately, generate an openapi spec from an existing collection. We only support the generation of requests from an openapi spec, not the other way around, at the moment.

**Reply by gabrielgrazziani** [reply]
wanted to use git integration

**Reply by develohpanda** [reply]
It is possible to move all your requests and request folders into a design document after creating it separately, but it's a manual process at the moment
